# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1507481268 -7200
#      Sun Oct 08 18:47:48 2017 +0200
# Node ID d52aa3a2794a075b7d010718ccff87c200288efd
# Parent  4e97a92d16c0cd6660d52fc32fa0642955525547
fix link underline position in gui terminal

* TerminalView.cpp (paintFilters): underline extended by 1 pixel to
  the right and ignoring font descends

diff --git a/libgui/qterminal/libqterminal/unix/TerminalView.cpp b/libgui/qterminal/libqterminal/unix/TerminalView.cpp
--- a/libgui/qterminal/libqterminal/unix/TerminalView.cpp
+++ b/libgui/qterminal/libqterminal/unix/TerminalView.cpp
@@ -1201,28 +1201,28 @@ void TerminalView::paintFilters(QPainter
                        endColumn*_fontWidth - 1, (line+1)*_fontHeight - 1 );
 
           // Underline link hotspots
           if ( spot->type() == Filter::Link ||
                spot->type() == Filter::ErrorLink)
             {
               QFontMetrics metrics(font());
 
-              // find the baseline (which is the invisible line that the characters in the font sit on,
-              // with some having tails dangling below)
-              int baseline = r.bottom() - metrics.descent();
+              // find the baseline (which is the invisible line that the
+              // characters in the font sit on
+              int baseline = r.bottom() + 1;
               // find the position of the underline below that
               int underlinePos = baseline + metrics.underlinePos();
 
               if (r.contains (mapFromGlobal(QCursor::pos())))
                 {
                   if (spot->type () == Filter::ErrorLink)
                     painter.setPen (QColor (255,0,0));
                   painter.drawLine (r.left(), underlinePos,
-                                    r.right(), underlinePos);
+                                    r.right() + 2, underlinePos);
                 }
             }
           // Marker hotspots simply have a transparent rectanglular shape
           // drawn on top of them
           else if ( spot->type() == Filter::Marker )
             {
               //TODO - Do not use a hardcoded colour for this
               painter.fillRect(r,QBrush(QColor(255,0,0,120)));
